File: rust/src/score_of_a_string_3110.rs
use crate::Solution;
impl Solution {
    pub fn score_of_string(s: String) -> i32 {
        let mut tmp: Vec<i32> = s
            .chars()
            .map(|key| key.to_ascii_lowercase() as i32)
            .collect();

        let n = tmp.len();
        let mut out = 0;
        for i in 0..n - 1 {
            out += (tmp[i] - tmp[i + 1]).abs();
        }
        out
    }
}

File: elixir/score_of_a_string_3110.ex
defmodule Solution do
  @spec score_of_string(s :: String.t) :: integer
  def score_of_string(s) do
    arr = String.to_charlist(s)
    Enum.reduce(arr |> tl, {0, arr |> hd} fn ch, {ans, prev} ->
      {ans + abs(ch - prev), ch}
    end) |> elem(0)
  end
end
